lips/lip-10.md

```diff
@@ -0,0 +1,190 @@
+
+## UI flow
+* Receiver shares their Libra ID with sender
+* Sender enters receiver’s Libra ID(and other txn details such as amount) in their wallet app
+* Sender’s wallet can fetch profile picture and full name of the receiver prior to sending money, and show it to sender, to improve security (if receiver VASP allows it)
+* Sender presses ‘send’ button
+* Receiver receives money and information about the sender (optionally)
+
+## Backend flow
+In order to support peer-2-peer payments, VASPs need to implement two Libra ID endpoints:
+* **Info endpoint** for querying optional information about the receiver. VASPs can provide this endpoint to improve UX and security of payment, but they are not obligated to - VASPs may choose not to share any information about the user via this endpoint, including whether the user exists or not
+* **Payment endpoint** to perform actual money transfer.
+
+Common considerations:
+
+* For both endpoints, all communications are signed via the same mechanism as for other off-chain communications, see the [Payloads section of lip-1](https://github.com/libra/lip/blob/master/lips/lip-1.mdx#payloads) for details. As such, all vasp-2-vasp communications are authenticated.
+
+* All Libra ID endpoints are nested under `pay.<version>` URI path:
+
+     `https://<hostname>:<port>/pay.<protocol_version>/<endpoint>`
+
+* All Libra ID endpoints use the same `base_url` and same `compliance_public_key` for authentication as other off-chain APIs.
+
+* All Libra ID endpoints are HTTP POST endpoints, they accept JSON-encoded requests and return JSON-encoded responses. Other encoding methods can be considered in the future, `Content-Type` and `Accept` HTTP headers can be used to define request/response formats.
+
+## Info endpoint
+
+```
+POST https://<hostname>:<port>/pay.<protocol_version>/info
+```
+
+**Info endpoint** is used by the **Sender** to request information about the **Receiver** before submitting an actual payment.
+
+Info endpoint takes both sender and receiver Libra IDs as arguments. Receiver VASP might decide to share more(or less) information based on sender ID, for example if the wallet supports some kind of contact list, they might decide to share more information if the sender is in the contact list of the receiver.
+
+**Request:**
+```json
+{
+   "sender_id": "<Sender Libra ID>",
+   "receiver_id": "<Receiver Libra ID>"
+}
+```
+
+**Response:**
+```json
+{
+   "receiver": <User Object>
+}
+```
+
+* Receiver Libra ID domain should belong to the Receiver VASP. Otherwise, Receiver VASP should return the `400` error code.
+* Sender Libra ID should have a Libra ID domain that belongs to Sender VASP. If receiver VASP makes a decision on how much information to share, based on the sender Libra ID, it should enforce this property and return 400 error code, if VASP tries to impersonate a user from Libra ID domain that it does not own.
```

Comment:
```suggestion
* Sender Libra ID must have a Libra ID domain that belongs to Sender VASP. If this is not the case, the Receiver VASP should return the `400` error code 
* Receiver VASP may optionally choose how much information to share based on the sender Libra ID. If Receiver VASP decides to not share data, it should return `400` error code.
```

Reply:
I edited it to split it out for how I believe you intended it to read, but I actually think that we shouldn't return the same error code for "i don't want to share information" that we do for "this person doesn't exist".  In one of those cases, you can still choose to send money to them

Reply:
Tried to rephrase it. I think if VASP don't want to share info, but user exists, they can just return empty user object(e.g. not sharing any fields). Otherwise return error